browser/tor/tor_profile_manager_browsertest.cc

```diff
@@ -322,6 +322,9 @@ IN_PROC_BROWSER_TEST_F(TorProfileManagerExtensionTest,
   EXPECT_TRUE(extensions::util::IsIncognitoEnabled(component_extension->id(),
                                                    tor_profile));
 
+  // TODO(darkdh): Try to resolve GetAllRelatedProfiles caused DCHECK failed.
```

Comment:
I'm a bit concerned that if we're getting a DCHECK here that we potentially have a bigger problem.

Reply:
cc: @darkdh

Reply:
@darkdh should confirm but I believe this affects test-only code, although I agree it would be good not to have to disable this subtest.

Reply:
yes, that is test-only code. The code block is installing a "not allowed" extension and intentionally enable "Allow in private" which is basically impossible in production, because users won't able to toggle the hidden "Allow in private" in that case. 
<img width="763" alt="Screen Shot 2021-04-28 at 10 38 35" src="https://user-images.githubusercontent.com/11330831/116448407-f0a02380-a80d-11eb-8816-64ee99a0eb57.png">

And here is the code stack @mariospr shared with me
```
[3474639:3474639:0323/175805.504137:FATAL:extension_util.cc(65)] Check failed: IncognitoInfo::IsIncognitoAllowed(&extension). 
#0 0x7fb6cc9119e9 base::debug::CollectStackTrace()
#1 0x7fb6cc811d63 base::debug::StackTrace::StackTrace()
#2 0x7fb6cc831a93 logging::LogMessage::~LogMessage()
#3 0x7fb6cc83243e logging::LogMessage::~LogMessage()
#4 0x556e5186a9fd extensions::util::(anonymous namespace)::SetCorsOriginAccessListForExtensionHelper()
#5 0x556e5186a7fb extensions::util::SetCorsOriginAccessListForExtension()
#6 0x556e5377be38 extensions::PermissionsUpdater::NetworkPermissionsUpdateHelper::UpdatePermissions()
#7 0x556e5377e01d extensions::PermissionsUpdater::SetUsesDefaultHostRestrictions()
#8 0x556e5377e6ea extensions::PermissionsUpdater::InitializePermissions()
#9 0x556e5373a2a2 extensions::ExtensionService::CheckPermissionsIncrease()
#10 0x556e5186077d extensions::ExtensionRegistrar::AddExtension()
#11 0x556e53739a65 extensions::ExtensionService::AddExtension()
#12 0x556e5373b5b8 extensions::ExtensionService::FinishInstallation()
#13 0x556e5373a245 extensions::ExtensionService::AddNewOrUpdatedExtension()
#14 0x556e5373ad55 extensions::ExtensionService::OnExtensionInstalled()
#15 0x556e537117b2 extensions::CrxInstaller::ReportSuccessFromUIThread()
#16 0x556e51245bda base::internal::Invoker<>::RunOnce()
...
```

Reply:
let's remove that code block because it is testing an impossible scenario

Reply:
@darkdh Thanks for confirming this is a test-only concern. As for removing the code, it does sound good to me but I wonder what to do then with https://github.com/brave/brave-browser/issues/14902. Should we close it then?

Reply:
Pushed a fix removing that code to `cr91` in https://github.com/brave/brave-core/commit/d5d6be4d0408ec61a107c6b15b23c485f65ae92a

Reply:
yes, we can close the issue